aggregator/aggregator.go

```diff
@@ -436,14 +427,40 @@ func (agg *aggregator) checkMetricType(mu unaggregated.MetricUnion) error {
 	}
 }
 
-func (agg *aggregator) ownedShards() []*aggregatorShard {
-	agg.RLock()
-	ownedShards := make([]*aggregatorShard, len(agg.shardIDs))
-	for i, shardID := range agg.shardIDs {
-		ownedShards[i] = agg.shards[shardID]
+func (agg *aggregator) ownedShards() (owned, toClose []*aggregatorShard) {
+	agg.Lock()
+	defer agg.Unlock()
```

Comment:
nit: seems this function can only returns at the end of the function body, perhaps close the lock explicitly before the return statement to save the overhead of `defer` if this in the hot path?

Reply:
I did that initially but feels more explicit this way, thankfully this is only called once a tick which happens by default every 10 minutes.

Reply:
Sounds good to me, I actually prefer `defer` myself so if it's called infrequently I like this approach.